src/Main.java

```diff
@@ -1,4 +1,94 @@
+            }
+            monthNumber++;
+        }
+        System.out.println("Task 02");
+        int i = 0;
+        while (i < 10) {
+            System.out.print(++i + " ");
+        }
+        System.out.println();
+        for (; i >= 1; i--) {
+            System.out.print(i + " ");
+        }
+        System.out.println("\nTask 03");
+        int yPopulation = 12_000_000;
+        int bornPerThousand = 17;
+        int diedPerThousand = 8;
+        int year = 0;
+        while (year < 10) {
+            yPopulation = yPopulation / 1000 * (1_000 + bornPerThousand - diedPerThousand);
+            System.out.println("Через " + ++year + " лет численность населения страны Y будет равна " + yPopulation);
+        }
+        System.out.println("\nЧерез " + year + " лет численность населения страны Y будет равна " + yPopulation);
+        System.out.println("Task 04");
+        float vasiliyDeposit = 15_000f;
+        int vasiliyGoal = 12_000_000;
+        int monthlyPercentage = 7;
+        monthNumber = 0;
+        while (vasiliyDeposit < vasiliyGoal) {
+            vasiliyDeposit = vasiliyDeposit / 100 * (100 + monthlyPercentage);
+            monthNumber++;
+            System.out.println("Месяц " + monthNumber + ", сумма накоплений Василия равна " + vasiliyDeposit + " рублей.");
+        }
+        System.out.println("Task 05");
+        monthNumber = 0;
+        vasiliyDeposit = 15_000;
+        while (vasiliyDeposit < vasiliyGoal) {
+            vasiliyDeposit = vasiliyDeposit * (100 + monthlyPercentage) / 100;
+            monthNumber++;
+            if (monthNumber % 6 == 0) {
+                System.out.println("Месяц " + monthNumber + ", сумма накоплений Василия равна " + vasiliyDeposit + " рублей.");
+            }
+        }
+        System.out.println("Task 06");
+        monthNumber = 0;
+        vasiliyDeposit = 15_000;
+        int goalYears = 9;
+        while (monthNumber < goalYears * 12) {
+            vasiliyDeposit = vasiliyDeposit * (100 + monthlyPercentage) / 100;
+            monthNumber++;
+            if (monthNumber % 6 == 0) {
+                System.out.println("Месяц " + monthNumber + ", сумма накоплений Василия равна " + vasiliyDeposit + " рублей.");
+            }
+        }
+        System.out.println("Task 07");
+        int firstFridayDate = 3;
+        int daysInMonth = 31;
+        for (int date = firstFridayDate; date <= daysInMonth; date = date + 7) {
+            System.out.println("Сегодня пятница, " + date + "-е число. Необходимо подготовить отчет.");
+        }
+        System.out.println("Task 08");
+        int firstCometAppearance = 0;
+        int cometPeriod = 79;
+        int currentYear = 2024;
+        int minimalYear = currentYear - 200;
+        int maximalYear = currentYear + 100;
+        //solution 1, 301 iterations
+        for (year = minimalYear; year <= maximalYear; year++) {
+            if (year % cometPeriod == firstCometAppearance % cometPeriod) {
```

Comment:
```suggestion
            if (year % cometPeriod == firstCometAppearance) {
```
зачем у второго брать остаток тоже? не нужно

Reply:
На случай, если год первого наблюдения кометы будет численно больше, чем периодичность появления.

Reply:
Ну такого не может быть, потому что иначе это будет уже не первый год пролёта кометы

Reply:
Нулевой год, как в условиях, тоже вряд ли первый, будто бы комета до Р.Х. не пролетала ни разу.
Допустим, записей/исторических документов о предыдущих пролётах не сохранилось/не велось